src/pages/workspaces/workspace/delete-workspace-modal.state.test.ts

```diff
@@ -0,0 +1,250 @@
+    })
+    jest.spyOn(console, 'log').mockImplementation(() => {
+    })
+  })
+
+  it('can initialize state for a google workspace with running apps', async () => {
+    // Arrange
+    const mockApps: Partial<AjaxAppsContract> = {
+      listWithoutProject: jest.fn()
+    }
+    asMockedFn((mockApps as AjaxAppsContract).listWithoutProject).mockResolvedValue([{
+      appName: 'app1',
+      status: 'running'
+    }])
+
+    const mockGetAcl = jest.fn().mockResolvedValue({ acl: { 'example1@example.com': {} } })
+    const mockGetBucketUsage = jest.fn().mockResolvedValue({ usageInBytes: 1234 })
+    const mockWorkspaces: DeepPartial<AjaxWorkspacesContract> = {
+      workspace: () => ({
+        getAcl: mockGetAcl,
+        bucketUsage: mockGetBucketUsage
+      })
+    }
+
+    const mockAjax: Partial<AjaxContract> = {
+      Apps: mockApps as AjaxAppsContract,
+      Workspaces: mockWorkspaces as AjaxWorkspacesContract
+    }
+    asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract)
+
+    // Act
+    const {
+      result,
+      waitForNextUpdate
+    } = renderHook(() => useDeleteWorkspaceState(googleWorkspace, mockOnDismiss, mockOnSuccess))
+    await waitForNextUpdate()
+
+    // Assert
+    expect(result.current.hasApps()).toBe(true)
+    expect(result.current.isDeleteDisabledFromResources).toBe(false)
+    expect(result.current.collaboratorEmails).toEqual(['example1@example.com'])
+    expect(result.current.workspaceBucketUsageInBytes).toBe(1234)
+  })
+
+  it('can initialize state for an azure workspace', async () => {
+    // Arrange
+    const mockListAppsV2: Partial<AjaxAppsContract> = {
+      listAppsV2: jest.fn()
+    }
+    asMockedFn((mockListAppsV2 as AjaxAppsContract).listAppsV2).mockResolvedValue([{
+      appName: 'example',
+      status: 'running'
+    }])
+
+    const mockWsmControlledResources: Partial<AjaxWorkspaceManagerContract> = {
+      controlledResources: jest.fn()
+    }
+    asMockedFn((mockWsmControlledResources as AjaxWorkspaceManagerContract).controlledResources).mockResolvedValue({
+      resources: [
+        { metadata: { name: 'example' } },
+        { metadata: { name: 'example' } }
+      ]
+    })
+
+    const mockAjax: Partial<AjaxContract> = {
+      Apps: mockListAppsV2 as AjaxAppsContract,
+      WorkspaceManagerResources: mockWsmControlledResources as AjaxWorkspaceManagerContract
+    }
+    asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract)
+
+    // Act
+    const {
+      result,
+      waitForNextUpdate
+    } = renderHook(() => useDeleteWorkspaceState(azureWorkspace, mockOnDismiss, mockOnSuccess))
+    await waitForNextUpdate()
+
+    // Assert
+    expect(result.current.hasApps()).toBe(true)
+    expect(result.current.isDeleteDisabledFromResources).toBe(true)
+    expect(result.current.controlledResourcesExist).toBe(true)
+  })
+
+  it('can delete an azure workspace', async () => {
+    // Arrange
+    const mockDelete = jest.fn().mockResolvedValue([])
```

Comment:
I'm not thrilled about the amount of ajax mocking I'm doing in most of these test cases. I played around with introducing a helper `mockAjaxCalls` but the complexity in making a generic fn that'd cover both azure and google codepaths didn't seem worth it for now. Happy to revisit if folks think otherwise.

Reply:
see my comments above.... WorkspaceProvider pattern could be a big help here.